lib/razorpay/payment.rb

```diff
@@ -27,10 +27,23 @@ def refund(options = {})
       self.class.request.post "#{id}/refund", options
     end
 
+    def refund!(options = {})
+      refund = refund options
+      refunded_payment = self.class.request.fetch id
+      @attributes = refunded_payment.attributes
+      refund
+    end
+
     def capture(options)
       self.class.request.post "#{id}/capture", options
     end
 
+    def capture!(options)
```

Comment:
Does a single capture make more sense? (since there are no extra network requests being made here.

Reply:
Still a change in behavior though, that's why I was hesitant.

Reply:
Okay, let us stick to the same rule in that case going forward (and add documentation as well)

>If an entity gets modified as a result of the API call, only the ! method will return the modified entity.

Reply:
Both will _return_ the modified entity. But only the ! method will modify the calling instance itself.
Will add to docs.

Reply:
Add docs and :shipit:

Reply:
Docs added